signac/contrib/project.py

```diff
@@ -1526,11 +1518,42 @@ def init_project(cls, name, root=None, workspace=None, make_dir=True):
             configuration.
 
         """
+        if args:
+            if len(args) == 1:
+                warnings.warn(
```

Comment:
If we add this API layer for backwards compatibility, we definitely need to test it well (and mark the tests as `TODO: Remove when...`). It appears that this isn't being tested right now, according to codecov.

Reply:
That's only showing up right now because as you pointed out I was duplicating the error handling in the `init_project` function, which is typically what is called in our tests (`signac.init_project` rather than `Project.init_project`). That being said, currently this behavior is being tested by a large number of tests that are still using the old API. As I mentioned in the description, I'll need to go through and update all the existing tests now that a reviewer has approved this approach, and when I do that I can add dedicated tests for this behavior.